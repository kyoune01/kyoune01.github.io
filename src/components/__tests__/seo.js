import React from "react"
import renderer from "react-test-renderer"
import SEO from "../seo"
import * as Gatsby from "gatsby"

const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      author: "Florian",
      description: "My description",
      title: "My Title",
    },
  },
}))

describe("SEO", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SEO title="Home" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
