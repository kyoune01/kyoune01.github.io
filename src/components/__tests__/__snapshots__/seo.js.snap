// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO renders correctly 1`] = `
[
  <title>
    Home | My Title
  </title>,
  <meta
    content="My description"
    name="description"
  />,
  <meta
    content="Home | My Title"
    property="og:title"
  />,
  <meta
    content="My description"
    property="og:description"
  />,
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Florian"
    name="twitter:creator"
  />,
  <meta
    content="Home | My Title"
    name="twitter:title"
  />,
  <meta
    content="My description"
    name="twitter:description"
  />,
]
`;
